src/transformers/modeling_utils.py

```diff
@@ -1058,6 +1059,15 @@ def base_model(self) -> nn.Module:
         """
         return getattr(self, self.base_model_prefix, self)
 
+    def can_generate(self) -> bool:
+        """
+        Returns whether this model can generate sequences with `.generate()`.
+
+        Returns:
+            :bool: Whether this model can generate sequences with `.generate()`.
+        """
+        return hasattr(self, "prepare_inputs_for_generation")
```

Comment:
Are we 100% sure all generative models will have this method in the future (and have it no)? cc @patrickvonplaten

Reply:
Yes every model needs this. Actually @gante could we add a `def prepare_inputs_for_generation(self, *args, **kwargs): raise NotImplementedError` to the `GenerationMixin`?

Reply:
BTW note that this check here is very similar: https://github.com/huggingface/transformers/blob/d0c1ded5f36e27cd74728c0127add5afdf1f2afa/src/transformers/generation/utils.py#L949 and probably outdated. 

We could maybe improve this check using this method here. => If a model has no `prepare_inputs_for_generation`, it cannot generate